packages/nextjs-contentful-renderer/src/getAffectedSlugs/getAffectedSlugs.ts

```diff
@@ -0,0 +1,81 @@
+import { getBlocksByModelIds } from '../getBlocksByModelIds';
+import { getPagesByBlockIds } from '../getPagesByBlockIds';
+import { getReplicasByPageIds } from '../getReplicasByPageIds';
+import { getRoutesByVariantIds } from '../getRoutesByVariantIds';
+import { getPagesByFooterIds } from '../getPagesByFooterIds';
+import { getPagesByThemeIds } from '../getPagesByThemeIds';
+import { getPagesByNavbarIds } from '../getPagesByNavbarIds';
+import { getVariantsByPageIds } from '../getVariantsByPageIds';
+import { ContentType, ContentTypeEntry } from './types';
+
+const nextContentTypeMap: Partial<Record<ContentType, ContentType>> = {
+  [ContentType.Model]: ContentType.Block,
+  [ContentType.Block]: ContentType.Page,
+  [ContentType.Page]: ContentType.Variant,
+  [ContentType.Theme]: ContentType.Page,
+  [ContentType.Navbar]: ContentType.Page,
+  [ContentType.Footer]: ContentType.Page,
+};
+
+export async function getAffectedSlugs(
+  { contentType, entryId, slug }: ContentTypeEntry,
+  preview: boolean = false
+): Promise<string[]> {
+  if (contentType === ContentType.Route || contentType === ContentType.Replica) {
+    return slug ? [slug] : [];
+  }
+
+  const slugs = new Set<string>();
+  await recursivelyGetSlugs(entryId, contentType, slugs, preview);
+  return Array.from(slugs);
+}
+
+async function recursivelyGetSlugs(
+  entryId: string,
+  contentType: ContentType,
+  slugs: Set<string>,
+  preview: boolean
+): Promise<void> {
```

Comment:
is it possible to change this method to return the slugs instead of modifying an argument?